### DISCUSSION TITLE
Switch lang not working with custom routes.

### DISCUSSION THREAD

**Discussion by mrFANRA** [nuxt-modules/i18n#2122] [Q&A] [answered]
Hi.

Why switch lang not working with custom routes? /en prefix in url showing page not found.

Reproduction:
[https://stackblitz.com/edit/nuxt-starter-waedfc](https://stackblitz.com/edit/nuxt-starter-waedfc)

**Comment by simkuns** [comment]
One way to solve this is to rename `sections` folder to `pages`, and delete `app/router.options.ts`.

**Reply by mrFANRA** [reply]
Yes, this way is working, but I need custom routes =(

**Reply by simkuns** [reply, marked as answer]
The issue stems from the fact that `@nuxtjs/i18n` utilizes the `extendPages` hook to retrieve the list of registered routes only during the build phase. However, app/router.options.ts is served as is once the server is already up and running, hence the mismatch.

If you really want to have `app/router.options.ts` then you will have to add routes manually.
```ts
export default {
  routes: () => [
    {
      path: '/',
      name: 'index___ru',
      component: () => import('~/sections/index.vue'),
    },
    {
      path: '/en',
      name: 'index___en',
      component: () => import('~/sections/index.vue'),
    },
  ],
};
```

Here's an example on how you can use the pages:extend hook within your configuration (https://v8.i18n.nuxtjs.org/guide/extend-pages/):
```ts
export default defineNuxtConfig({
  modules: [
   (_, nuxt) => {
      nuxt.hook('pages:extend', (pages) => {
        pages.push({
          name: 'index',
          path: '/',
          file: '~/sections/index.vue',
        });
      });
    },
    '@nuxtjs/i18n'
  ],
  // ...
});
```

**Reply by mrFANRA** [reply]
Thank you very much, it is work! 

Is it temporary solution for beta version or for release too?
